lib/service/ean.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:dio/dio.dart';
+
+class EanService {
+  Future<Map> fetchInformation(String eanCode) async {
+    try {
+      Response response = await Dio().post("https://gtin.hoelshare.com/$eanCode");
+
+      if (response.statusCode != 200) {
+        throw 'The response code is not valid';
```

Comment:
du kannst das try catch sparen wenn du das hier direkt wieder faengst und dann returns

Hier kann genau so gut direkt die fehler message returned werden in einer map

Reply:
ich dachte nur weil du den fehler hier am ende der methode catched und dann returnst kannst du den auch einfach direkt hier returnen
Wahrscheinlich wuerde es auch zu einem Fehler kommen wenn die gecatchete exception returned wird weil das keine Map ist und der Return type der Funktion map ist, oder?

Reply:
Ich lasse das try catch global weg und werfe dann den Fehler. Dann kann in die Methode, welche die Funktion benutzt, im Catch den Fehler abfangen